test/widgets/message_list_test.dart

```diff
@@ -212,9 +212,9 @@ void main() {
 
     testWidgets('show names on DMs', (tester) async {
       await setupMessageListPage(tester, messages: [
-        eg.dmMessage(from: eg.selfUser, to: []),
-        eg.dmMessage(from: eg.otherUser, to: [eg.selfUser]),
-        eg.dmMessage(from: eg.thirdUser, to: [eg.selfUser, eg.otherUser]),
+        eg.dmMessage(id: 1, from: eg.selfUser, to: []),
+        eg.dmMessage(id: 2, from: eg.otherUser, to: [eg.selfUser]),
+        eg.dmMessage(id: 3, from: eg.thirdUser, to: [eg.selfUser, eg.otherUser]),
```

Comment:
msglist test [nfc]: Add distinct ids to messages in 'show names on DMs' setup
    
    This will prevent a future test failure when other parts of
    the app begin assuming that messages have unique ids.

Cool, that's helpful.

As I suggested above, though, it seems like we probably need to do this more broadly. I've just gone and implemented that, as #343. So you can stack this PR atop that one (even before it's merged), and then this isn't needed.

Similarly #343 will make unnecessary several other uses of `id` on `eg.streamMessage` and `eg.dmMessage` calls in these tests, just like the ones in main now that it removes.